src/main/java/it/pagopa/ecommerce/commons/client/NpgClient.java

```diff
@@ -603,11 +606,16 @@ private NpgResponseException exceptionToNpgResponseException(
 
                 errors = responseErrors.stream()
                         .map(error -> GatewayError.valueOf(error.getCode())).toList();
+                statusCode = Optional.of(e.getStatusCode());
             } catch (IOException ex) {
                 String errorMessage = "Invalid error response from NPG with status code %s";
                 log.error(errorMessage.formatted(e.getStatusCode()));
 
-                return new NpgResponseException(errorMessage.formatted(e.getStatusCode()), ex);
+                return new NpgResponseException(
+                        errorMessage.formatted(e.getStatusCode()),
+                        Optional.of(e.getStatusCode()),
```

Comment:
```suggestion
                        statusCode,
```

Reply:
If an exception is thrown `statusCode` is stille set to `Optional.empty()`
I could assign status code before invoking `objectMapper.readValue()` but I feel it would make the code a bit brittle

Reply:
Yep, you're right, let leave it as is 😊